tests/nftables/test_nftables.py

```diff
@@ -0,0 +1,191 @@
+import ipaddress
+import subprocess
+import unittest
+import json
+import os
+import sys
+from time import sleep
+from typing import List
+from ipaddress import ip_address
+
+from tests.nftables import mock_lapi
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+os.chdir(SCRIPT_DIR)
+
+
+def run_cmd(cmd: List[str], trace_error=True):
+    with subprocess.Popen(cmd, stdout=subprocess.PIPE) as p:
+        if p.wait() and trace_error:
+            print(cmd, "exited with non-zero code with following logs")
+            print(p.stdout.read().decode())
+            sys.exit(1)
+        else:
+            return p.stdout.read().decode()
+
+
+def generate_n_decisions(n: int, action="ban", dup_count=0, ipv4=True):
+    if dup_count >= n:
+        print("generate_n_decisions got dup_count>=n")
```

Comment:
print + exit(1) is a sign that maybe the output is intended for stderr. Ask yourself: is it useful in a pipe, in a script? Or is it a message/warning/error for a human? In the latter case, return the message with an exception. If you don't want or need to create your exceptions, I think "raise SystemExit(message)" is safe to use and sets $? to 1.